src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -226,6 +285,15 @@ def __str__(self):
         }
         return pformat(o)
 
+    def toJSON(self):
+        return {'name': self.name, "endpoints": self.endpoints.__dict__, "suffixes": self.suffixes.__dict__}
+
+    @classmethod
+    def fromJSON(cls, json_str):
```

Comment:
what if the json file corrupt, please give a proper error message to help the user to recover.